messaging/broker.go

```diff
@@ -92,25 +126,67 @@ func (b *Broker) Open(path string, u *url.URL) error {
 	}
 	b.path = path
 
-	// Require a non-blank connection address.
-	if u == nil {
-		return ErrConnectionAddressRequired
+	// Ensure root directory exists.
+	if err := os.MkdirAll(path, 0700); err != nil {
```

Comment:
As @rothrock suggested, I updated all directories to `0777` and all files to `0666` and we can let `umask` handle it. 96748cb